### PR TITLE
Km rna with umis methods

### PR DESCRIPTION
**Pull request by kayleemathews**

### CODE REVIEW

File: website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+The genome-aligned BAM file was then used to calculate summary metrics using RNASeQC and Picard's CollectRNASeqMetrics and CollectMultipleMetrics tools. CollectMultipleMetrics was used with the programs “CollectInsertSizeMetrics” and “CollectAlignmentSummaryMetrics”. 
+
+The final outputs of the RNA with UMIs pipeline included a transcriptome-aligned, duplicate-marked BAM file with corresponding BAI and duplicate metrics files, a genome-aligned, duplicate-marked BAM file with corresponding BAI and duplicate metrics files, and metrics files generated with RNASeQC and Picard tools.
```
Comment (ekiernan) [inline website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md:21]:
```suggestion
The final outputs of the RNA with UMIs pipeline included a transcriptome-aligned, duplicate-marked BAM file with corresponding index and duplication metrics, a genome-aligned, duplicate-marked BAM file with corresponding index, duplication metrics, and metrics generated with RNASeQC and Picard tools.
```

**Comment by takutosato** [comment]
Sure thing, I will take a look.

File: website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
...ads, --outSAMtype was set to "BAM Unsorted" and --outSAMunmapped was set to "Within". A transcriptome-aligned BAM was also output with --quantMode = TranscriptomeSAM. To match [ENCODE bulk RNA-seq data standards](https://www.encodeproject.org/data-standards/rna-seq/long-rnas/), the alignment was performed with parameters --outFilterType = BySJout, --outFilterMultimapNmax = 20, --outFilterMismatchNmax = 999, --alignIntronMin = 20, --alignIntronMax = 1000000, --alignMatesGapMax = 1000000, --alignSJoverhangMin = 8, and --alignSJDBoverhangMin = 1. The fraction of reads required to match the reference was set with --outFilterMatchNminOverLread = 0.33 and the fraction of allowable mismatches to read length was set with --outFilterMismatchNoverLmax = 0.1. Chimeric alignments were included with --chimSegmentMin = 15, where 15 was the minimum length of each segment, and --chimMainSegmentMultNmax = 1 to prevent main chimeric segments from mapping to multiple sites. To output chimeric segments with soft-clipping in the aligned BAM, --chimOutType was set to "WithinBAM SoftClip". A maximum of 20 protruding bases at the ends of alignments was allowed with --alignEndsProtrude = 20 ConcordantPair.
```
[the remark this answers is not in the document]
Comment (takutosato) [reply website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md:15]: Adding to the last sentence: "A maximum of 20 protruding bases at the ends of alignments was allowed with --alignEndsProtrude = 20 ConcordantPair, so the reads from small cDNA fragments (where we read into adapters) will not be dropped." I'd be happy to show you what this means in our meeting later.

File: website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+Following alignment, the read group information from the genome-aligned BAM was copied to the transcriptome-aligned BAM using Samtools. Both BAM files were then sorted by coordinate with Picard's SortSam tool, and index and MD5 digest files were created. Duplicate reads were sorted by UMI with UMI-tools to differentiate between biological and PCR duplicates. To specify the tag where the UMIs were stored, --extract-umi-method was set to "tag" and --umi-tag was set to "RX". Unmapped reads were deduplicated with --unmapped-reads = use. Tagged BAM files were output using the option --output-bam. Picard's SortSam tool was used again to sort the BAM files by queryname and create MD5 digest files. PCR duplicates were marked, and duplicate metrics were calculated using Picard's MarkDuplicates tool with GATK v4.1.9.0. BAM files were sorted with Picard's SortSam tool, and index and MD5 digest files were created.
```
Comment (takutosato) [inline website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md:17]: I think we can omit the first sentence about read group information—it's an implementation detail that normal users don't need to know about.

File: website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+The genome-aligned, duplicate-marked BAM file was then used to calculate summary metrics using RNASeQC, Picard's CollectRNASeqMetrics and CollectMultipleMetrics tools, and GATK's (v4.2.5.0) GetPileupSummaries and CalculateContamination tools. CollectMultipleMetrics was used with the programs “CollectInsertSizeMetrics” and “CollectAlignmentSummaryMetrics”. GetPileupSummaries was run with the read filters, "WellformedReadFilter" and "MappingQualityAvailableReadFilter" disabled.
```
Comment (takutosato) [inline website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md:19]: "GetPileupSummaries was run with the read filters, "WellformedReadFilter" and "MappingQualityAvailableReadFilter" disabled."—probably another implementation detail we can omit here. Although first I should ask, who's the intended reader?

Comment (takutosato) [reply website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md:19]: "GATK's GetPileupSummaries and CalculateContamination tools" -> "GATK's CalculateContamination." GetPileupSummaries is just a preprocessing step.

File: website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md (hunk 3 above)
Comment (takutosato) [reply website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md:17]: "Duplicate reads were sorted by UMI…" -> "Using umi-tools group, putative duplicates were further divided into subgroups by UMIs and any sequencing errors in UMI bases were corrected."

"Unmapped reads were deduplicated with --unmapped-reads = use." What this argument turned on the unmapped reads are included in the output file as is, rather than discarded. This is important for us, so if the user chooses to reprocess the data in the future, we have not thrown out any reads.

Comment (takutosato) [reply website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md:17]: Let's talk about this paragraph later.

**Review by takutosato — COMMENTED** [review]
@kayleemathews @ekiernan here is my review, sorry for the delay!

File: website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md (hunk 3 above)
Comment (jessicaway) [inline website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md:17]: I don't expect you to know the answer to this, but is there a reason we are using GATK v4.1.9.0 here? This should be updated to v4.2.5.0 to address the log4j vulnerability (that may very well be in a separate PR though)

**Review by jessicaway — APPROVED** [review]
Looks good!

File: website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md (hunk 3 above)
Comment (takutosato) [reply website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md:17]: Great catch. No reason, and in fact we should be using a picard docker here. I will correct this my contamination PR.

File: website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md (hunk 4 above)
Comment (kayleemathews) [reply website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md:19]: I hesitate to remove GetPileupSummaries - to me it seems like an important detail for getting the same results.

File: website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kayleemathews) [reply website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md]: Updated

File: website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md (hunk 3 above)
Comment (kayleemathews) [reply website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md]: Updated

File: website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md (hunk 4 above)
Comment (takutosato) [reply website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md]: Yea actually the read filters for GetPileupSummaries are important, so good call!

File: website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md [hunk 5]
```diff
@@ -12,9 +12,9 @@ Data preprocessing, gene counting, and metric calculation were performed using t
...
...ligned BAM was also output with --quantMode = TranscriptomeSAM. To match [ENCODE bulk RNA-seq data standards](https://www.encodeproject.org/data-standards/rna-seq/long-rnas/), the alignment was performed with parameters --outFilterType = BySJout, --outFilterMultimapNmax = 20, --outFilterMismatchNmax = 999, --alignIntronMin = 20, --alignIntronMax = 1000000, --alignMatesGapMax = 1000000, --alignSJoverhangMin = 8, and --alignSJDBoverhangMin = 1. The fraction of reads required to match the reference was set with --outFilterMatchNminOverLread = 0.33 and the fraction of allowable mismatches to read length was set with --outFilterMismatchNoverLmax = 0.1. Chimeric alignments were included with --chimSegmentMin = 15, where 15 was the minimum length of each segment, and --chimMainSegmentMultNmax = 1 to prevent main chimeric segments from mapping to multiple sites. To output chimeric segments with soft-clipping in the aligned BAM, --chimOutType was set to "WithinBAM SoftClip". A maximum of 20 protruding bases at the ends of alignments was allowed with --alignEndsProtrude set to "20 ConcordantPair" to prevent reads from small cDNA fragments that were sequenced into adapters from being dropped.
```
Comment (takutosato) [inline website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md:15]: As we discussed earlier this week `--outFilterMatchNminOverLread` and `--outFilterMismatchNoverLmax` are equivalent. I still need to confirm this, will update the doc when I confirm that this is not an issue. We will likely set --outFilterMatchNminOverLread to default.

File: website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md [hunk 6]
```diff
@@ -8,15 +8,15 @@ Below we provide an example methods section for publications using the RNA with
...
+Following alignment, both BAM files were sorted by coordinate with Picard's SortSam tool. UMI-tools was then used to further divide putative duplicates into subgroups based on UMI and sequencing errors in UMIs were corrected. To specify the tag where the UMIs were stored, --extract-umi-method was set to "tag" and --umi-tag was set to "RX". Unmapped reads were included in the output file with --unmapped-reads = use. Tagged BAM files were output using the option --output-bam. SortSam was used again to sort the BAM files by queryname for Picard's MarkDuplicates tool with GATK. MarkDuplcates was used to mark PCR duplicates and calculate duplicate metrics. After duplicate marking, BAM files were sorted by coordiante using SortSam to facilitate downstream analysis.
```
Comment (takutosato) [inline website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md]: "Picard's MarkDuplicates tool with GATK" can omit "with GATK"

Comment (takutosato) [reply website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md]: But other than that great!

**Comment by takutosato** [comment]
@kayleemathews this looks great, thank you so much for doing this. Requested one small change but otherwise good to go.

File: website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md (hunk 5 above)
Comment (kayleemathews) [reply website/docs/Pipelines/RNA_with_UMIs_Pipeline/rna-with-umis.methods.md:15]: I'll update once we know for sure that it works! Thanks!
